src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -23,89 +28,168 @@ import Data.HashMap.Lazy qualified as HashMap (toList)
-                , branchOf src
-                ]
-            _ -> Just $ podVersion pod
-
-      -- add edges between spec and specdeps
-      traverse_ (edge pkg . PodfilePkg . depName) $ podSpecs pod
-      -- add a label for version
-      label pkg $ PodfileVersion pkgVersion
+    addSpec Pod{podName, podVersion, podDeps} = do
+      -- Add edges to transitive dependencies.
+      traverse_ (edge pkg . PodfilePkg . depName) podDeps
+      -- Label pods with versions.
+      label pkg $ PodfileVersion podVersion
+      where
+        pkg = PodfilePkg podName
+
+-- We use this cache to avoid repeated invocations since gtraverse may revisit
+-- nodes. In practice, this saves quite a bit of time (e.g. on example projects,
+-- it has historically turned ~2 minute analyses into ~15 seconds).
+type PodSpecCache = Map Text PodSpecJSON
+
+buildGraph ::
+  (Has ReadFS sig m, Has Exec sig m, Has Diagnostics sig m) =>
+  Path Abs File ->
+  PodLock ->
+  m (Graphing Dependency)
+buildGraph lockFilePath lockFile@PodLock{lockExternalSources} =
+  -- If `pod` _is_ installed, then we can shell out to `pod ipc spec` to
+  -- convert locally vendored `.podspec` files to JSON and read them for
+  -- their Git locators. We do this because locally vendored `.podspec`
+  -- files are often vendored specifically because they are not available on
+  -- the Cocoapods registry.
+  evalState @PodSpecCache mempty $ gtraverse replaceVendoredPods staticGraph
+  where
+    staticGraph :: Graphing Dependency
+    staticGraph = buildGraphStatic lockFile
+
+    lockFileDir :: Path Abs Dir
+    lockFileDir = parent lockFilePath
+
+    replaceVendoredPods ::
+      ( Has Exec sig m
+      , Has Diagnostics sig m
+      , Has (State PodSpecCache) sig m
+      ) =>
+      Dependency ->
+      m Dependency
+    replaceVendoredPods d@Dependency{dependencyType, dependencyName} = case dependencyType of
+      PodType -> case Map.lookup dependencyName lockExternalSources of
+        Just es -> case es of
+          ExternalPodSpec podSpecPath ->
+            readPodSpecAt podSpecPath <||> pure d
+          ExternalPath podPath ->
+            readPodSpecAt (toText $ toString podPath </> (toString dependencyName <.> "podspec")) <||> pure d
+          ExternalGitType _ -> pure d
+          ExternalOtherType -> pure d
+        Nothing -> pure d
+      _ -> pure d
+
+    readPodSpecAt ::
+      ( Has Exec sig m
+      , Has Diagnostics sig m
+      , Has (State PodSpecCache) sig m
+      ) =>
+      Text ->
+      m Dependency
+    readPodSpecAt podSpecPath = context ("Resolving vendored podspec at " <> showT podSpecPath) $ do
+      podSpecCache :: PodSpecCache <- get
+      (PodSpecJSON ExternalGitSource{urlOf, tagOf, commitOf, branchOf}) <-
+        case Map.lookup podSpecPath podSpecCache of
+          Just cached -> pure cached
+          Nothing -> do
+            podSpec <-
+              execJson
+                lockFileDir
+                Command
```

Comment:
[Optional] Although this command is only used here, I would move it outside of `buildGraph` scope, so it follows the stylistic convention of other analyzers. 

[Optional] Add ref URL in comment - https://guides.cocoapods.org/terminal/commands.html#pod_ipc_spec